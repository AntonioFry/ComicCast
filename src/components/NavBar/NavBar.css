
nav {
  background-color: black;
  padding: 10px 0px 0px;
  width: 100%;
}

h1 {
  color: white;
  font-size: 2em;
  margin: 0px;
  text-align: center;
}

.navigation-links-container {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  margin: 10px 0px 0px;
}

.navigation-links {
  align-items: center;
  color: white;
  display: flex;
  font-size: 1.25em;
  height: 30px;
  justify-content: center;
  text-align: center;
  text-decoration: none;
}

.navigation-links:hover {
  background-color: white;
  color: black;
}