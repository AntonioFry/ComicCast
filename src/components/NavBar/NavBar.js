import React from 'react';
import { NavLink } from 'react-router-dom';
import './NavBar.css';
import { setSearchResult } from '../../actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const NavBar = ({ setSearchResult }) => {
  return (
    <nav>
      <h1>ComicCast</h1>
      <div className="navigation-links-container">
        <NavLink className="navigation-links" to="/" onClick={() => setSearchResult("")}>Home</NavLink>
        <NavLink className="navigation-links" to="/Favorites" onClick={() => setSearchResult("")}>Favorites</NavLink>
        <NavLink className="navigation-links" to="/Search">Search</NavLink>
      </div>
    </nav>
  )
}

const mapDispatchToProps = (dispatch) => ({
  setSearchResult: search => dispatch(setSearchResult(search))
});

NavBar.propTypes = {
  setSearchResult: PropTypes.func.isRequired
}

export default connect(null, mapDispatchToProps)(NavBar);