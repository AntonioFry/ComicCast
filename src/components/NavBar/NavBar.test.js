import React from 'react'
import { NavBar } from './NavBar'
import { shallow } from 'enzyme';

describe('NavBar', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  })
})