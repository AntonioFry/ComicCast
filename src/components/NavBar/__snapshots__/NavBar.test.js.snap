// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should match the snapshot 1`] = `
<nav>
  <h1>
    ComicCast
  </h1>
  <div
    className="navigation-links-container"
  >
    <NavLink
      className="navigation-links"
      to="/"
    >
      Home
    </NavLink>
    <NavLink
      className="navigation-links"
      to="/Search"
    >
      Search
    </NavLink>
  </div>
</nav>
`;
